app/api/api_v1/routers/admin.py

```diff
@@ -349,48 +422,108 @@ def start_ingest(db: Session, s3_client: S3Client, file_contents: str):
+                "Write Bulk Document Ingest CSV to S3 Failed.",
+                extra={
+                    "props": {
+                        "superuser_email": current_user.email,
+                    }
+                },
+            )
+            raise HTTPException(
+                status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+                detail="Unexpected error, fail to write Bulk Document Ingest CSV to S3",
+            )
+        if (
+            type(result_events) is bool
+        ):  # S3Client returns False if the object was not created
+            _LOGGER.error(
+                "Write Bulk Event Ingest CSV to S3 Failed.",
+                extra={
+                    "props": {
+                        "superuser_email": current_user.email,
+                    }
+                },
+            )
+            raise HTTPException(
+                status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+                detail="Unexpected error, fail to write Bulk Event Ingest CSV to S3",
+            )
+        else:
+            documents_csv_s3_location = str(result_documents.url)
+            events_csv_s3_location = str(result_events.url)
+            _LOGGER.info(
+                "Write Event Ingest CSV complete.",
+                extra={
+                    "props": {
+                        "superuser_email": current_user.email,
+                        "documents_csv_s3_location": documents_csv_s3_location,
+                        "events_csv_s3_location": events_csv_s3_location,
+                    }
+                },
+            )
+    except Exception as e:
+        _LOGGER.exception(
+            "Unexpected error, writing Bulk Document Ingest CSV content to S3",
+            extra={"props": {"errors": str(e)}},
+        )
+        raise HTTPException(status_code=status.HTTP_500_INTERNAL_SERVER_ERROR) from e
+
+    # PHASE 3 - Start the ingest (kick off background task to do the actual ingest)
+    background_tasks.add_task(
+        _start_ingest,
+        db,
+        s3_client,
+        s3_prefix,
+        documents_file_contents,
+        events_file_contents,
     )
 
-    # PHASE 2 - Ingest
-    # Start the background task to do the actual ingest.
-    background_tasks.add_task(start_ingest, db, s3_client, file_contents)
-
     _LOGGER.info(
-        "Background Bulk Ingest Task added",
+        "Background Bulk Document/Event Ingest Task added",
         extra={
             "props": {
                 "superuser_email": current_user.email,
-                "csv_s3_location": csv_s3_location,
+                "documents_csv_s3_location": documents_csv_s3_location,
+                "events_csv_s3_location": events_csv_s3_location,
             }
         },
     )
 
     # TODO: Add some way the caller can monitor processing pipeline...
-
-    return csv_s3_location
+    return BulkImportResult(
+        import_s3_prefix=s3_prefix,
+        detail=None,  # TODO: add detail?
```

Comment:
We could put the detail from the validation result on here

Reply:
ah, right, have looked at this. I agree we could put this detail in, but probably something for a follow up because `detail` here has a very specific type & expected content.